api/types/swarm/container.go

```diff
@@ -66,10 +66,12 @@ type ContainerSpec struct {
 	// The format of extra hosts on swarmkit is specified in:
 	// http://man7.org/linux/man-pages/man5/hosts.5.html
 	//    IP_address canonical_hostname [aliases...]
-	Hosts     []string            `json:",omitempty"`
-	DNSConfig *DNSConfig          `json:",omitempty"`
-	Secrets   []*SecretReference  `json:",omitempty"`
-	Configs   []*ConfigReference  `json:",omitempty"`
-	Isolation container.Isolation `json:",omitempty"`
-	Sysctls   map[string]string   `json:",omitempty"`
+	Hosts            []string            `json:",omitempty"`
+	DNSConfig        *DNSConfig          `json:",omitempty"`
+	Secrets          []*SecretReference  `json:",omitempty"`
+	Configs          []*ConfigReference  `json:",omitempty"`
+	Isolation        container.Isolation `json:",omitempty"`
+	Sysctls          map[string]string   `json:",omitempty"`
+	MemorySwap       int64               `json:",omitempty"`
```

Comment:
Does this make the default `0` for swap? I think the current default will be "unlimited", or is it `0` already? (didn't try yet)

Reply:
Perhaps we can use a pointer here, so that `nil` is "unset" (equal to `-1` for the container API)??

Reply:
Given how it's then fed to the daemon's executor in daemon/cluster/executor/container/container.go (https://github.com/moby/moby/pull/37872/files#diff-9651468cafd3eee09570f3733efd7dc4), prior to this patch we just create an empty `enginecontainer.Resources{}` struct (https://github.com/moby/moby/blob/v17.03.2-ce/daemon/cluster/executor/container/container.go#L400), so its `MemorySwap` field is 0, and we never change it.

So the current behaviour is already defaulting to 0.